interfaces/builtin/vcio.go

```diff
@@ -0,0 +1,55 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// https://docs.broadcom.com/doc/12358545
+// https://github.com/raspberrypi/linux/tree/rpi-5.4.y/drivers/char/broadcom
+const vcioSummary = `allows access to VideoCore I/O`
+
+// The raspberry pi allows programming its GPU from userspace via the /dev/vcio
+// device. These operations should be considered privileged since the driver
+// assumes trusted input, therefore require manual connection.
+const vcioBaseDeclarationSlots = `
+  vcio:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const vcioConnectedPlugAppArmor = `
+# Description: Can access to vcio.
+
+# vcio v1 access to ARM VideoCore (BCM_VCIO) for userspace GPU programming
+/dev/vcio rw,
```

Comment:
do we need udev rules to add this vcio device node to the device cgroup of the snap?

Reply:
@anonymouse64 - I checked on a raspberry pi and the devices were already loaded. I suggest adding them only if we get bug reports.

Reply:
Sorry, I was thinking about the kmod backend, not udev. Yes we need a udev entry. Thanks!

Reply:
Done (verified the added rule correctly tags on rpi2).